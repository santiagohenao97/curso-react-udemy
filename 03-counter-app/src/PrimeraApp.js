//Functionals components FC
import React from 'react';
import PropTypes from 'prop-types'

const PrimeraApp = ({saludo = 'Hola por defecto', subtitulo}) =>{ 
    //una forma de enviar props por defecto

    /*const saludito  = 'Hola Mundo '//No imprime booleanos, ni objetos directamente
    const objeto  = {
        nombre: "santiago",
        edad: 24
    }*/

    return (
        <>
            <h1>{ saludo }</h1>
            <h3>{ subtitulo }</h3>
            {/*<h3>{ saludito }</h3>
            <pre> {JSON.stringify(objeto) }</pre>*/}
            <p>Mi primera aplicaci√≥n</p>
        </>
    );
}

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired
}

PrimeraApp.defaultProps = { //Es otra forma de enviar props por defecto
    subtitulo: 'Soy un subtitulo'
}

export default PrimeraApp
