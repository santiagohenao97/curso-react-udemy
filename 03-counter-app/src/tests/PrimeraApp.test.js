import { getByText, render } from '@testing-library/react';
import { shallow } from 'enzyme';
import React from 'react';
import PrimeraApp from '../PrimeraApp'

describe('Pruebas en <PrimeraApp />', () => {
    
    /*test('debe mostrar el mensaje Hola soy Gokú', () => {
        const saludo = 'Hola, soy Gokú';
        const {getByText} = render(<PrimeraApp saludo={saludo}/>)

        expect(getByText(saludo)).toBeInTheDocument()
    })*/

    //Usando Enzyme
    test('debe mostrar el mensaje Hola soy Gokú', () => {
        const saludo ='Hola, soy Gokú'
        const wrapper = shallow(<PrimeraApp saludo = {saludo}/>)
        expect( wrapper ).toMatchSnapshot()
        
    })

    test('debe mostrar un subtitulo', () => {
        const saludo ='Hola, soy Gokú'
        const subtitulo = 'Hola soy un subtitulo'

        const wrapper = shallow(<PrimeraApp saludo = {saludo} subtitulo={subtitulo}/>)
        
        const textoParrafo = wrapper.find('h3').text()
        console.log(textoParrafo)
        expect(textoParrafo).toBe(subtitulo)
        
    })
})
