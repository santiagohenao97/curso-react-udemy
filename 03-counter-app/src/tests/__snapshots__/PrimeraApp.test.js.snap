// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> debe mostrar el mensaje Hola soy Gokú 1`] = `
<Fragment>
  <h1>
    Hola, soy Gokú
  </h1>
  <h3>
    Soy un subtitulo
  </h3>
  <p>
    Mi primera aplicación
  </p>
</Fragment>
`;
